tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/typecast/TypeCastExpressionsTest.java

```diff
@@ -183,6 +183,13 @@ public void testDirectlyUnmatchedUnionToUnionCastNegativeTwo() {
         BRunUtil.invoke(result, "testDirectlyUnmatchedUnionToUnionCastNegative_2");
     }
 
+    @Test(expectedExceptions = BLangRuntimeException.class,
+            expectedExceptionsMessageRegExp = "error: \\{ballerina\\}TypeCastError \\{\"message\":\"incompatible " +
+                    "types: 'map<json>' cannot be cast to 'Bar'.*")
+    public void testMutableJsonMappingToExclusiveRecordNegative() {
```

Comment:
We can use trap and assert this in b7a itself?

Reply:
+1, Let's improve this test case like that.

Reply:
Updated the test case.